frontend/packages/console-dynamic-plugin-sdk/src/app/modal-support/ModalProvider.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+
+type CloseModal = () => void;
+
+export type ModalComponent<T = {}> = () => React.FC<
+  {
+    closeModal: CloseModal;
+  } & T & { isOpen?: boolean }
+>;
+
+export type LaunchModal = (component: ModalComponent, extraProps: any) => void;
+
+type ModalProps = {
+  launchModal: LaunchModal;
+  closeModal: CloseModal;
+};
+
+const defaultModalContext: ModalProps = {
+  launchModal: null,
+  closeModal: null,
+};
+
+export const ModalContext = React.createContext(defaultModalContext);
+
+export const ModalProvider: React.FC = ({ children }) => {
+  const [isOpen, setOpen] = React.useState(false);
+  const [Component, setComponent] = React.useState<ReturnType<ModalComponent>>(null);
+  const [componentProps, setComponentProps] = React.useState<{}>({});
```

Comment:
can we use `unknown` here? `{}` is not an appropriate type here. Might have to be `any`?

Reply:
`unknown` would be better.